boa/src/builtins/value/operations.rs

```diff
@@ -1,4 +1,78 @@
 use super::*;
+use crate::Interpreter;
+use std::borrow::Borrow;
+
+#[allow(clippy::float_cmp)]
+/// https://tc39.es/ecma262/#sec-numeric-types-number-equal
+fn strict_number_equals<T: Into<f64>>(a: T, b: T) -> bool {
+    let a: f64 = a.into();
+    let b: f64 = b.into();
+
+    if a.is_nan() || b.is_nan() {
+        return false;
+    }
+
+    a == b
```

Comment:
I think that `f64::NAN == f64::NAN` is defined as `false` in Rust. so this would suffice:
```suggestion
    a.into() == b.into()
```

Reply:
done